hazelcast/src/hazelcast/client/network.cpp

```diff
@@ -112,11 +116,10 @@ namespace hazelcast {
             }
 
             bool ClientConnectionManagerImpl::start() {
-                std::lock_guard<std::mutex> guard(lock);
-                if (alive) {
-                    return true;
+                bool expected = false;
+                if (!alive.compare_exchange_strong(expected, true)) {
+                    return false;
                 }
-                alive.store(true);
 
```

Comment:
Do you intent to change the behavior here, or just use atomics instead of locks?
Previously if the thing was already alive (alive == true), the function returned true; but now it returns false.